basic_nodes/src/test/java/club/mondaylunch/gatos/basicnodes/test/GetAtIndexNodeTest.java

```diff
@@ -0,0 +1,141 @@
+        }
+
+        Map<String, DataBox<?>> input = Map.of(
+            "input", ListDataType.GENERIC_LIST.create(testArrayList),
+            "index", DataType.NUMBER.create(0.0)
+        );
+
+        Map<String, DataBox<?>> input2 = Map.of(
+            "input", ListDataType.GENERIC_LIST.create(testArrayList),
+            "index", DataType.NUMBER.create(1.0)
+        );
+
+        var output = BasicNodes.GET_AT_INDEX.compute(input, Map.of(), Map.of());
+        Assertions.assertEquals(0, output.get("output").join().value());
+        Assertions.assertEquals(output.get("output").join().type(), DataType.ANY);
+
+        var output2 = BasicNodes.GET_AT_INDEX.compute(input2, Map.of(), Map.of());
+        Assertions.assertEquals(1, output2.get("output").join().value());
+    }
+
+    @Test
+    public void correctlyEvaluatesReturnType() {
+        List<Integer> testArrayList = new ArrayList<>();
+
+        for (int i = 0; i < 10; i++) {
+            testArrayList.add(i);
+        }
+
+        Map<String, DataBox<?>> input = Map.of(
+            "input", ListDataType.GENERIC_LIST.create(testArrayList),
+            "index", DataType.NUMBER.create(0.0)
+        );
+
+        Map<String, DataType<?>> inputTypes = Map.of(
+            "input_type", DataType.NUMBER
+        );
+
+        var output = BasicNodes.GET_AT_INDEX.compute(input, Map.of(), inputTypes);
+        Assertions.assertEquals(0, output.get("output").join().value());
+        Assertions.assertEquals(output.get("output").join().type(), DataType.NUMBER);
+    }
+
+    @Test
+    public void correctlyEvaluatesReturnTypeWithWrongInput() {
+        Map<String, DataBox<?>> input = Map.of(
+            "input", ListDataType.GENERIC_LIST.create(new ArrayList<>()),
+            "index", DataType.NUMBER.create(0.0)
+        );
+
+        Map<String, DataType<?>> inputTypes = Map.of(
+            "input_type", DataType.NUMBER
+        );
+
+        var output = BasicNodes.GET_AT_INDEX.compute(input, Map.of(), inputTypes);
+        Assertions.assertEquals(OptionalDataType.GENERIC_OPTIONAL, output.get("output").join().value());
+        Assertions.assertEquals(output.get("output").join().type(), DataType.NUMBER);
+    }
+
+    @Test
+    public void correctlyEvaluatesLinkedLists() {
+        List<String> testLinkedList = new LinkedList<>();
+
+        for (int i = 0; i < 10; i++) {
+            testLinkedList.add("i" + i);
+        }
+
+        Map<String, DataBox<?>> input = Map.of(
+            "input", ListDataType.GENERIC_LIST.create(testLinkedList),
+            "index", DataType.NUMBER.create(0.0)
+        );
+
+        Map<String, DataType<?>> inputTypes = Map.of(
+            "input_type", DataType.STRING
+        );
+
+        var output = BasicNodes.GET_AT_INDEX.compute(input, Map.of(), inputTypes);
+        Assertions.assertEquals("i0", output.get("output").join().value());
+        Assertions.assertEquals(output.get("output").join().type(), DataType.STRING);
+    }
+
```

Comment:
There should be a test here that involves connecting the node in a graph, to test the dynamic typing.

Reply:
Currently, there is no node with an output of a list. How would you suggest doing this test?

Reply:
Uh yeah IDK what I was thinking there. See the suggestion above, that's the only thing I think that's missing from the test.